test/jackdaw/serdes/resolver_test.clj

```diff
@@ -98,6 +98,29 @@
                     (.serialize (.serializer resolved) "avro-topic")
                     (.deserialize (.deserializer resolved) "avro-topic"))))))
 
+    (testing "avro reader serdes with optional schema"
+      (let [resolver-fn (resolver/serde-resolver :schema-registry-url ""
+                                                 :schema-registry-client (reg/mock-client))
+            writer-avro-config {:serde-keyword :jackdaw.serdes.avro.confluent/serde
+                                :schema-filename "resources/example_schema.avsc"
+                                :key? false}
+            ;; by providing :serde/type of :read-only we can allow a serde without local schema
+            reader-avro-config {:serde-keyword :jackdaw.serdes.avro.confluent/serde
+                                :key? false
+                                :serde-type :read-only}
```

Comment:
I would rather have this work like `:key?`.

Example:
```
{:serde-keyword :jackdaw.serdes.avro.confluent/serde
 :key? false
 :read-only? true}
```

Or if `:key?` is not required:
```
{:serde-keyword :jackdaw.serdes.avro.confluent/serde
 :read-only? true}
```